speechbrain/utils/data_utils.py

```diff
@@ -374,15 +373,12 @@ def update_to(self, b=1, bsize=1, tsize=None):
                     dest_unpack = os.path.dirname(dest)
                 print(f"Extracting {dest} to {dest_unpack}")
                 # shutil unpack_archive does not work with tar.gz files
-                if (
-                    source.endswith(".tar.gz")
-                    or source.endswith(".tgz")
-                    or source.endswith(".gz")
-                ):
-                    out = dest.replace(".gz", "")
-                    with gzip.open(dest, "rb") as f_in:
-                        with open(out, "wb") as f_out:
-                            shutil.copyfileobj(f_in, f_out)
+                if source.endswith((".tar.gz", ".tgz", ".gz")):
+                    import tarfile
+
+                    tar = tarfile.open(dest, "r:gz")
+                    tar.extractall(dest_unpack, filter="data")
```

Comment:
Not OK, `filter` was added in Python 3.12, c.f. https://docs.python.org/3/library/tarfile.html#tarfile.TarFile.extractall

Reply:
I guess this could just be removed but this seems iffy from a security perspective. Could just document it though.